src/Service/CalculatingManager.php

```diff
@@ -111,4 +103,39 @@ private function globalCost(array $housings): float
     {
         return $this->getHourlyRateRepo->findOneBy([])->getRate() * $this->totalTime($housings);
     }
+
+    /**
+     * Compute profits for each condo
+     * @param array $housings
+     * @return array
+     */
+    public function profitabilityCondo(array $housings): array
+    {
+        $condoProfit = [];
+        foreach ($housings as $housing) {
+            $activities = $housing->getHousingActivities();
+            $hoursTotal = $this->getHourActivities($activities);
+
+            $profit = $housing->getFee() - ($hoursTotal * $this->getHourlyRateRepo->findOneBy([])->getRate());
```

Comment:
avant ton foreach, fait un `$hourlyRate = $this->getHourlyRateRepo->findOneBy([])->getRate();`
car là tu refais la requete en BDD pour chaque tour de boucle alors que la valeur du $rate ne change pas